src/main/java/org/betonquest/betonquest/database/MySQL.java

```diff
@@ -154,35 +155,40 @@ private void migration2(final Connection connection) throws SQLException {
                     + "CHANGE COLUMN playerID profileID CHAR(36) NOT NULL, "
                     + "MODIFY COLUMN pointer VARCHAR(255) NOT NULL, " + "ADD FOREIGN KEY (profileID) REFERENCES "
                     + prefix + "profile (profileID) ON DELETE CASCADE");
+            statement.executeUpdate("DELETE FROM " + prefix + "objectives WHERE id NOT IN "
+                    + "(SELECT MIN(id) FROM " + prefix + "objectives GROUP BY playerID, objective)");
             statement.executeUpdate("ALTER TABLE " + prefix + "objectives "
                     + "CHANGE COLUMN playerID profileID CHAR(36) NOT NULL, "
                     + "MODIFY COLUMN objective VARCHAR(510) NOT NULL, "
                     + "MODIFY COLUMN instructions VARCHAR(2046) NOT NULL, " + "DROP PRIMARY KEY," + "DROP COLUMN id");
-            statement.executeUpdate("ALTER IGNORE TABLE " + prefix + "objectives "
-                    + "ADD PRIMARY KEY (profileID, objective)");
+            statement.executeUpdate("ALTER TABLE " + prefix + "objectives ADD PRIMARY KEY (profileID, objective)");
             statement.executeUpdate("ALTER TABLE " + prefix + "objectives "
                     + "ADD FOREIGN KEY (profileID) REFERENCES " + prefix + "profile (profileID) ON DELETE CASCADE");
```

Comment:
You can combine these (and similar) into one big `ALTER TABLE` statement:
```java
            statement.executeUpdate("ALTER TABLE " + prefix + "objectives "
                    + "CHANGE COLUMN playerID profileID CHAR(36) NOT NULL, "
                    + "MODIFY COLUMN objective VARCHAR(510) NOT NULL, "
                    + "MODIFY COLUMN instructions VARCHAR(2046) NOT NULL, "
                    + "DROP PRIMARY KEY, "
                    + "DROP COLUMN id,"
                    + "ADD PRIMARY KEY (profileID, objective), "
                    + "ADD FOREIGN KEY (profileID) REFERENCES " + prefix + "profile (profileID) ON DELETE CASCADE");
```

Reply:
done

Reply:
welly that was half, there's also another one above each of them that can be merged too.

Reply:
Is it now fine?